File: app/src/main/java/com/eugene_poroshin/money_manager/fragments/AddCategoryDialogFragment.java
package com.eugene_poroshin.money_manager.fragments;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.eugene_poroshin.money_manager.R;

public class AddCategoryDialogFragment extends DialogFragment {

    private EditText editText;

    public interface EditNameDialogListener {
        void onFinishEditDialog(String inputText);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.dialog_fragment_add_category, container, false);
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        builder.setTitle("Добавление новой категории");
        builder.setView(inflater.inflate(R.layout.dialog_fragment_add_category, null))
                .setPositiveButton("Добавить", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int id) {
                        sendBackResult();
                    }
                })
                .setNegativeButton("Отмена", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int id) {
                        if (dialog != null) {
                            dialog.dismiss();
                        }
                    }
                });
        return builder.create();
    }

    public void sendBackResult() {
        editText = getDialog().findViewById(R.id.edit_text_category_name);
        editText.requestFocus();
        getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
        EditNameDialogListener listener = (EditNameDialogListener) getTargetFragment();
        listener.onFinishEditDialog(editText.getText().toString());
        dismiss();
    }
}

File: app/src/main/java/com/eugene_poroshin/money_manager/fragments/AddCategoryDialogFragment.kt
package com.eugene_poroshin.money_manager.fragments

import android.app.Dialog
import android.os.Bundle
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.core.os.bundleOf
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.setFragmentResult
import com.eugene_poroshin.money_manager.R

class AddCategoryDialogFragment : DialogFragment(R.layout.dialog_fragment_add_category) {

    private var editText: EditText? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireActivity())
        builder.setTitle("Добавление новой категории")
        builder.setView(requireActivity().layoutInflater.inflate(R.layout.dialog_fragment_add_category, null))
            .setPositiveButton("Добавить") { _, _ -> sendBackResult() }
            .setNegativeButton("Отмена", null)
        return builder.create()
    }

    private fun sendBackResult() {
        //todo Fragment Result Listener - is it ok?
        editText = dialog?.findViewById(R.id.edit_text_category_name)
        val result = editText?.text?.toString()?.capitalize()
        when (result) {
            "" -> Toast.makeText(context, "Введите название категории", Toast.LENGTH_SHORT).show()
            null -> Toast.makeText(context, "Введите название категории", Toast.LENGTH_SHORT).show()
            else -> {
                setFragmentResult("requestKey", bundleOf("bundleKey" to result))
                dismiss()
            }
        }
    }

    override fun onDestroyView() {
        editText = null
        super.onDestroyView()
    }
}